### PR TITLE
partially fixes #165 - created navbar, sidebars for storelisting page

### LINKED ISSUE #165: Create a store listing page for the apps
Presently clicking on any apps takes the user to the apps page. There should be an intermediate store listing page like google playstore where user will get various information about the app he/she has selected.
### proposed design of apps.loklak.org store listing page
* A left menu bar with the category to which the selected app belongs highlighted
* A right side bar containing similar apps (same category) for the ease of the user
* A middle section with details about the selected app. The middle section will contain basic app info like app name, author's name, app promo image, a carousel showing app screen shots, author contact info, and most importantly what the app does, how to use the app, tutorial to use the app (optional), how to deploy the app. 
These informations can be fetched from the app.json of each app. Since the app.json of the existing apps does not contains such information right now, so a sample app.json or dummy data can be used for testing the site.
 Once everything is working, the app.json of the existing apps can be updated. 
The page will also contain link to test the live app.

### PR DESCRIPTION
**Pull request by djmgit**
Partially fixes issue #165. Created navigation bar, left side bar
for categores, and right bar for similar app suggestion. Implemented
deeplinks for category filtering.
A url of form http://apps.loklak.org/#<category_name> will show only
the apps belonging to the mentioned category.
Storelisting page will have the following url.
http://apps.loklak.org/details.html?q=<app_name>
The page will show all the informaton listed for given app (app_name)
The category to which the selected app belongs will be hghlighted.
Clicking on any category name will take the user to starting page
showing only the filtered apps which belong to the selected category.
The middle section (presently empty) will hold the app information.

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by djmgit** [comment]
Test link : loklakapps.surge.sh/details?q=whatsTrending

**Comment by djmgit** [comment]
@kavithaenair @SKrPl open for review

**Comment by mariobehling** [comment]
I see a big white space on your test link. How this supposed to look like?
screenshot from 2017-06-02 18-09-14

**Comment by djmgit** [comment]
mariobehling as I have mentioned here - https://github.com/fossasia/apps.loklak.org/issues/165 the middle section which is presently blank now will contain the selected app information, it will be loadded dynamically from app.json file of the selected app, so that the users will no how to use the app, the developer of the app will have to provide all the info in app.json of the app. I will be creating a tutorial page where everything will be mentioned about creating a an (part of my GSOC project)

Here are the screenshots of the prototype on which I am working

mid1

mid2

Work is still left, so i will be sending another PR for the middle section so that PRs do not become too large. The middle section will be able to render markdown.

**Comment by djmgit** [comment]
@mariobehling since the present app.json files do not provide all the required information, so for now I have modified the app.json of whatsTrending app with dummy data for testing purpose. Once this page is ready, the app.json of the existing apps will be updated one by one, relevant issues will be created. The guidelines for creating a proper app.json file will be provided on the tutorial page which I will be creating soon.

File: details.html [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    </nav>
+    <div class="main app" ng-controller="detailsApp">
+      <div class="row">
+        <div class="col-md-3">
+          <div class="sidebar">
+            <div class="sidebar-header">
+              <div class="sidebar-header-image">
+                <img src = "images/sidebar-images/menu.png">
+              </div>
+              <div class="catgegory-header">
```
Comment (kavithaenair) [inline details.html]: Please fix the typos in the class name and give a appropriate name.

Comment (kavithaenair) [inline details.html]: I cannot find a page header here where it displays the category of the apps as in apps.loklak.org .

**Comment by djmgit** [comment]
@kavithaenair that is the ultimate target, when a an user clicks on any app, it will go to the store listing page, there the user will come to know everything about the app and then he/she can try the app using the link provided (Try now)

**Comment by djmgit** [comment]
Since the page is not ready right now and the app.json of all the apps are not ready so I have yet not integrated the storelisting page with all the apps, presently I am only using whatsTrending app for testing purpose (and dummy data as shown in the screenshots)

File: details.html (hunk 1 above)
Comment (djmgit) [inline details.html]: @kavithaenair the category to which the selected app belongs will be highlighted in the left category sidebar

Comment (djmgit) [inline details.html]: @kavithaenair the category to which the selected app belongs will be highlighted in the left category sidebar

**Comment by kavithaenair** [comment]
@djmgit while refreshing the page it loads slowly and also shows `{{category.name}}` in the categories navbar and `{{app.name}}` on the page (as images as loading slowly).

File: details.html (hunk 1 above)
Comment (kavithaenair) [inline details.html]: @djmgit  correct the typo here as well `Know more about ths category` on the category navbar.

**Comment by djmgit** [comment]
@kavithaenair I have removed 'know more about this category' as it was a place holder which I had forgotten to remove. 
I have moved the angularjs script to top of the page, hope this will prevent the angular variables from getting displayed on the screen. Regarding speed, I will try to optimize it more as the page develops.

**Comment by mariobehling** [comment]
@djmgit Ok, thanks for explaining. So, this is work in progress then. With the screenshot that you are showing I see that the diagram is scaled out of proportion. This is also the case on the current page. I would expect that you see this yourself and correct it when you work on this. Please fix issues like that on the go. Cheers.

File: css/details.css [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    margin-right: 20px;
+}
+
+.menu-list {
+    margin-left: 0px;
+    padding-left: 0px;
+}
+
+.menu-list-item {
+    text-decoration: none !important;
```
Comment (vibhcool) [inline css/details.css]: Could you prevent using !important as it leads to more errors in future. 
see this -> https://stackoverflow.com/questions/9245353/what-does-important-in-css-mean

**Comment by djmgit** [comment]
@vibhcool removed !important

File: app_list.js [hunk 3]
```diff
@@ -34,6 +42,9 @@ app.controller("app_list", function ($scope, $http) {
 
 
     $scope.categoryFilter = function (event) {
+        if ($scope.category !== null) {
```
Comment (Achint08) [inline app_list.js]: Can't this condition be removed and let `$scope.category` be made null directly?

**Comment by djmgit** [comment]
@Achint08 thanks for pointing out, removed the null checking condition

**Pull request merged**